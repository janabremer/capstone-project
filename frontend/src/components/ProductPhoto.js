import styled from 'styled-components/macro'

export default function ProductPhoto({url, photographer}) {
    return(
        <ImgStyled src={url} alt="" width="300px"/>
    )
}

const ImgStyled = styled.img`
    width: 100%;
    height: auto;
    border-radius: 5px;  
`
