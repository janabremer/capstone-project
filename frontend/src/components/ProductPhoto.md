```js
<ProductPhoto
  url={
    'https://images.pexels.com/photos/102104/pexels-photo-102104.jpeg?auto=compress&cs=tinysrgb&h=350'
  }
  photographer={'mali maeder'}
/>
```
