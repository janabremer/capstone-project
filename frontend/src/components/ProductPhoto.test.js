import { render } from '@testing-library/react'
import 'jest-styled-components'
import ProductPhoto from './ProductPhoto'

describe('ProductPhoto', () => {
    it('renders correctly', () => {
        const { container } = render(
            <ProductPhoto
                url={'https://images.pexels.com/photos/102104/pexels-photo-102104.jpeg?auto=compress&cs=tinysrgb&h=350'}
                photographer={'mali maeder'}
            />
        )
        expect(container.firstChild).toMatchSnapshot()
    })

    it('displays image', () => {
        const { getByRole } = render(
            <ProductPhoto
                url={'https://images.pexels.com/photos/102104/pexels-photo-102104.jpeg?auto=compress&cs=tinysrgb&h=350'}
                photographer={'mali maeder'}
            />
        )
        expect(getByRole('img', { url: 'https://images.pexels.com/photos/102104/pexels-photo-102104.jpeg?auto=compress&cs=tinysrgb&h=350' })).toBeInTheDocument()
    })
    
})