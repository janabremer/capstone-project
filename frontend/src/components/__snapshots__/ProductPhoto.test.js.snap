// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPhoto renders correctly 1`] = `
.c0 {
  width: 100%;
  height: auto;
  border-radius: 5px;
}

<img
  alt=""
  class="c0"
  src="https://images.pexels.com/photos/102104/pexels-photo-102104.jpeg?auto=compress&cs=tinysrgb&h=350"
  width="300px"
/>
`;
